app/src/main/kotlin/github/com/rhacco/dota2androidapp/lists/LiveMatchesAdapter.kt

```diff
@@ -27,6 +32,29 @@ class LiveMatchesAdapter(context: Context) : RecyclerView.Adapter<LiveMatchesVie
         return true
     }
 
+    fun setOfficialNames(proPlayers: List<ProPlayerEntity>) {
+        proPlayers.forEach {
+            var index = 0
+            while (index < mItemsData.size) {
```

Comment:
You have double loops, which will make the runtime a nightmare. I think you can make this easier by adding a query to the DAO to look up by steamid/account_id.

Reply:
Better yet, you could craft a SQL query to query by a list of ids and just use the results.

Something like SELECT * from proPlayers where id IN (listOfProPlayers) ?

Reply:
So `proPlayers` here is already a trimmed list that only has pro players which are in one of the displayed live matches. This is a maximum of 100 pros since at max 10 matches are added/handled at once with 10 players each. `mItemsData.size` is 10-20 and I'm also breaking the loops, so this code really shouldn't be an issue.

You may be referring to the other double loop I introduced in `ProPlayersLocalDataSource::checkProPlayersHelper` that trims the full list of pro players to the list used above:
```
playerSteamIds.forEach { playerSteamId ->
    proPlayers.forEach { proPlayer ->
        if (playerSteamId == proPlayer.account_id)
            result.add(proPlayer)
        }
}
```
While `proPlayers` is the full list of pro players from the OpenDota API which could end up being very large, the size of `playerSteamIds` is again 100 at max (same reason as above). I could see a reason to optimize that code but it shouldn't be O(n^2) in any case

Reply:
Ok I see what you were going for. This is fine for now. I might add my own optimization but at least the logic for matching the IDs is separated.